tests/virt_autotest/setup_dns_service.pm

```diff
@@ -0,0 +1,223 @@
+    assert_script_run("cp $dns_resolver_file $dns_resolver_file.orig");
+    assert_script_run("$add_resolver_nameserver");
+    assert_script_run("mv $dns_resolver_tmpfile $dns_resolver_file");
+    my $add_resolver_search = "sed -irn \'s/^search.*\$/& $dns_domain_forward/\' $dns_resolver_file";
+    assert_script_run("$add_resolver_search");
+
+    #Reset default-lease-time and max-lease-time in /etc/dhcpd.conf to 28800
+    my $set_dhcpd_lease_time = "sed -ir \"s/default-lease-time\.*\$/default-lease-time $dhcpd_lease_time;/g; s/max-lease-time.*\$/max-lease-time $dhcpd_lease_time;/g;\" $dhcpd_config_file";
+    assert_script_run("$set_dhcpd_lease_time");
+
+    my $get_os_installed_release = "lsb_release -r | grep -oE \"[[:digit:]]{2}\"";
+    my $os_installed_release     = script_output($get_os_installed_release, $wait_script, type_command => 1, proceed_on_failure => 1);
+    if ($os_installed_release gt '11') {
+        assert_script_run("systemctl enable $dhcpd_service_name");
+        assert_script_run("systemctl restart $dhcpd_service_name");
+        assert_script_run("systemctl status $dhcpd_service_name");
+    }
+    else {
+        assert_script_run("service $dhcpd_service_name restart");
+        assert_script_run("service $dhcpd_service_name status");
+    }
+
+    #Write {vmhostname => vmipaddress} key => value pairs into vm_hash_forward dictionary
+    #The dictionary will be used to fill out the forward dns zone file
+    #Write {vmipaddress last part => vmhostname} key => value pairs into vm_hash_reverse dictionary
+    #The dictionary will be used to fill out the reverse dns zone file
+    my %vm_hash_macaddress          = "";
+    my %vm_hash_forward             = "";
+    my %vm_hash_reverse             = "";
+    my $get_vm_hostnames_inactive   = "virsh list --inactive | grep sles | awk \'{print \$2}\'";
+    my $vm_hostnames_inactive       = script_output($get_vm_hostnames_inactive, $wait_script, type_command => 1, proceed_on_failure => 1);
+    my @vm_hostnames_inactive_array = split(/\n+/, $vm_hostnames_inactive);
+    my $get_vm_hostnames            = "virsh list  --all | grep sles | awk \'{print \$2}\'";
+    my $vm_hostnames                = script_output($get_vm_hostnames, $wait_script, type_command => 1, proceed_on_failure => 1);
+    my @vm_hostnames_array          = split(/\n+/, $vm_hostnames);
+    my $get_vm_macaddress           = "";
+    my @vm_macaddresses_array       = "";
+    my $get_vm_ipaddress            = "";
+    my $vm_ipaddress                = "";
+    my $vm_ipaddress_lastpart       = "";
+    my $vm_hash_index               = "";
+    foreach (@vm_hostnames_array)
+    {
+        if ($_ ~~ @vm_hostnames_inactive_array) {
+            assert_script_run("virsh start $_");
+        }
+        else {
+            assert_script_run("virsh reboot $_");
+        }
+    }
```

Comment:
Making all guests running at the same time may meet resource lackage issue on some resource limited HW,  which may make the start or reboot fail.
Besides do not forget to shutdown the vms when operations done.